### PR TITLE
fix(outline): apply filters to have outline for icon

### PR DESCRIPTION
**Pull request by invincibleJai**
- apply filters to have outline for icon

Screenshot :
Screenshot 2019-06-08 at 1 39 07 AM

### CODE REVIEW

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const KnativeDecorator: React.FunctionComponent<KnativeDecoratorTypes> = ({
+  x,
+  y,
+  width,
+  height,
+}) => {
+  const knativeDecorator = (
+    <g key="knative">
+      <defs>
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx]: Use the `SvgDefs` or `SvgDropShadowFilter` component

File: frontend/public/extend/devconsole/components/topology/shapes/WorkloadNode.tsx [hunk 2]
```diff
@@ -33,13 +33,12 @@ const WorkloadNode: React.FC<NodeProps<WorkloadData>> = ({
       onSelect={onSelect}
       attachments={[
         workload.data.isKnativeResource && (
-          <image
+          <KnativeDecorator
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/WorkloadNode.tsx]: I just realized that this shouldn't be a decorator. It should be a child of `BaseNode` that comes after the `PodStatus`. Please move it. Remove the `key` attribute at the same time.

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { getImageForIconClass } from '../../../../../components/catalog/catalog-item-icon';
+
+type KnativeDecoratorTypes = {
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx]:
```suggestion
type KnativeIconProps = {
```

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+const KnativeDecorator: React.FunctionComponent<KnativeDecoratorTypes> = ({
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx]: This shouldn't be a decorator. Rename accordingly:
```suggestion
const KnativeIcon: React.FC<KnativeIconProps> = ({
```

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  x,
+  y,
+  width,
+  height,
+}) => {
+  const knativeDecorator = (
+    <g key="knative">
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx]: Remove the `<g>` and replace with `<React.Fragment>` when you update your use of `<defs>` (see other comment.

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx (hunk 1 above)
Comment (invincibleJai) [reply frontend/public/extend/devconsole/components/topology/shapes/KnativeDecorator.tsx]: @christianvogt have made use of `SvgDropShadowFilter` and it provides a shadow/fade effect, to me fade looks better than the just border. Have updated both screenshots in the description.

**Review by karthikjeeyar — APPROVED** [review]
LGTM

**Comment by christianvogt** [comment]
I don't particularly like that shadow border. It needs a hard white  edge to start. It could fade a little maybe but the current shadow you provided looks too transparent.
fyi @serenamarie125

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-unused-vars, no-undef */
+import * as React from 'react';
+import { getImageForIconClass } from '../../../../../components/catalog/catalog-item-icon';
+import SvgDropShadowFilter from '../../../shared/components/svg/SvgDropShadowFilter';
+import { createSvgIdUrl } from '../../../shared/utils/svg-utils';
+
+type KnativeIconProps = {
+  id: string;
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeIcon.tsx]: The ID should not be a prop as it is used only for the filter reference which is a constant in the system.
Define a constant ID in this file please.
use `KnativeIconOutlineFilterId`

File: frontend/public/extend/devconsole/components/topology/shapes/KnativeIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+};
+
+const KnativeIcon: React.FC<KnativeIconProps> = ({ id, x, y, width, height }) => {
+  const knativeIcon = (
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/topology/shapes/KnativeIcon.tsx]: Remove the declaration and return the fragment immediately in the function without the `return` keyword.

**Comment by invincibleJai** [comment]
> I don't particularly like that shadow border. It needs a hard white edge to start. It could fade a little maybe but the current shadow you provided looks too transparent.

@christianvogt icon has white border, as it's thin and icon size is small in default zoom so not coming out but with shadow looks better. Can we tweek something in `SvgDropShadowFilter` or modify to make shadow bit hard to make it more intutive.

**Comment by christianvogt** [comment]
@invincibleJai if `SvgDropShadowFilter` doesn't fit your needs, simply use `SvgDefs` and create your own filter.

**Comment by invincibleJai** [comment]
@christianvogt have updated filter to have white border and verified in safari as well. updated description with a screenshot.

**Review by christianvogt — APPROVED** [review]

**Pull request merged**